### PR TITLE
Work on typing the types subsystem

### PR DESCRIPTION
**Pull request by pudo**
This is a first stab at introducing more type info to the ftm libraries. I'm trying to keep this incremental because by the very nature of it, it's going to touch a ton of files and go out of date quick :/

### CODE REVIEW

File: followthemoney/types/common.py [hunk 1]
```diff
@@ -51,21 +55,33 @@ class PropertyType(object):
...
+        text = sanitize_text(raw)
         if text is None:
             return None
-        return self.clean_text(text, **kwargs)
+        return self.clean_text(text, fuzzy=fuzzy, format=format, proxy=proxy)
 
-    def clean_text(self, text: str, **kwargs) -> Optional[str]:
+    def clean_text(
+        self,
+        text: str,
```
Comment (Rosencrantz) [inline followthemoney/types/common.py:80]: And here too, if feasible. Make the argument names consistent across the common definition and the overrides.

File: followthemoney/types/identifier.py [hunk 2]
```diff
@@ -1,4 +1,5 @@
 import re
+from typing import Callable, Sequence
```
Comment (Rosencrantz) [inline followthemoney/types/identifier.py:2]: This doesn't seem to be used

File: followthemoney/types/language.py [hunk 3]
```diff
@@ -85,7 +91,7 @@ class LanguageType(EnumType):
     LANGUAGES = get_env_list("FTM_LANGUAGES", LANGUAGES)
```
Comment (Rosencrantz) [inline followthemoney/types/language.py:91]: This is marked as constant coz it's all uppercase, but we're overwriting it on the line below. Is this legit?

**Comment by pudo** [comment]
I think I've got all the names in sync now. Opted for `value` for `validate()` because `value` gets used for all the other methods that accept an existing FtM prop value.

File: followthemoney/types/language.py (hunk 3 above)
Comment (pudo) [reply followthemoney/types/language.py:91]: `constant-ish`. It's sort of stable at run-time, is what I mean.

**Review by Rosencrantz — APPROVED** [review]

**Comment by pudo** [comment]
Merged via #479.

**Pull request closed without merging**